backend/app/adapter/db/url.go

```diff
@@ -170,6 +171,101 @@ func (u URLSql) composeParamList(numParams int) string {
 	return parameterStr
 }
 
+// FindURLsByUser finds all urls created by a given user
+func (u URLSql) FindURLsByUser(user entity.User) ([]entity.URL, error) {
+	statement := fmt.Sprintf(`
+SELECT %s, %s, %s, %s, %s
+FROM %s
+INNER JOIN %s ON %s.%s = %s.%s
+WHERE "%s"=$1;`,
+		table.URL.ColumnAlias,
+		table.URL.ColumnOriginalURL,
+		table.URL.ColumnExpireAt,
+		table.URL.ColumnCreatedAt,
+		table.URL.ColumnUpdatedAt,
+
+		table.URL.TableName,
+
+		table.UserURLRelation.TableName,
+
+		table.URL.TableName,
+		table.URL.ColumnAlias,
+		table.UserURLRelation.TableName,
+		table.UserURLRelation.ColumnURLAlias,
+
+		table.UserURLRelation.ColumnUserEmail,
+	)
+
+	urls := []entity.URL{}
+	rows, err := u.db.Query(statement, user.Email)
+
+	if err != nil {
+		return urls, nil
+	}
+
+	for rows.Next() {
+		var url entity.URL
+		err = rows.Scan(
+			&url.Alias,
+			&url.OriginalURL,
+			&url.ExpireAt,
+			&url.CreatedAt,
+			&url.UpdatedAt,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		urls = append(urls, url)
+	}
+
+	return urls, nil
+}
+
+// DeleteURLByUser removes URL having given alias created by given user
+func (u *URLSql) DeleteURLByUser(user entity.User, alias string) (entity.URL, error) {
```

Comment:
I wonder whether we are actually using this?

Reply:
Delete button is used in this PR: https://github.com/short-d/short/pull/615